controller/src/main/java/io/pravega/controller/store/stream/Scope.java

```diff
@@ -52,4 +85,26 @@
      */
     void refresh();
 
+    static String encodeStreamInScope(String name, long creationTime) {
```

Comment:
These 3 methods would look much nicer if encapsulated (non-statically) into their own class:
- Two constructors: one with name and creation time and one with encoded name. Both constructors would immediately encode and store in a field.
- The class would implement comparable, which is what you have in `compareStreamInScope`.


Another question? Where is this encoding going? Is it stored anywhere? If so, have you considered versioning it?

Reply:
@shrids please update the PR with latest from PR #3312 as these methods have been removed.